packages/connect/src/protocol-connect/validate-response.ts

```diff
@@ -46,12 +45,7 @@ export function validateResponse(
       `HTTP ${status}`,
       codeFromHttpStatus(status)
     );
-    if (
-      methodKind == MethodKind.Unary &&
-      parsedType &&
-      !parsedType.stream &&
-      !parsedType.binary
-    ) {
+    if (methodKind == MethodKind.Unary && parsedType && !parsedType.stream) {
```

Comment:
This is an unrelated fix. For Connect unary, we always try to parse an error from the response body if we receive a non-successful HTTP status - not just for JSON, but for protobuf binary as well. This has slipped through our test suite so far, but is covered now.